pkg/file/joiner/internal/job.go

```diff
@@ -110,6 +107,19 @@ func (j *SimpleJoinerJob) nextReference(level int) error {
 	chunkAddress := swarm.NewAddress(data[cursor : cursor+swarm.SectionSize])
 	err := j.nextChunk(level-1, chunkAddress)
 	if err != nil {
+		if err == io.EOF {
+			return err
+		}
+		// if the last write is a "dangling chunk" the data chunk will have been moved
+		// to an intermediate level. In this edge case, the error must be suppressed,
+		// and the cursor manually to data length boundary to terminate the loop in
+		// the calling frame.
+		if j.readCount+int64(len(data)) == j.spanLength {
+			j.cursors[level] = len(j.data[level])
+			err = j.sendChunkToReader(data)
+			return err
+		}
+		j.logger.Debugf("error in chunk join for %v: %v", chunkAddress, err)
```

Comment:
Error should not be returned and logged. I am suggesting to handle the error, and log it if necessary by the topmost caller in the execution stack. With both return and log, the same error may be logged multiple times.

Reply:
Ok I will _wrap_ the error then. The problem was the store conceals the detail; it just says `not found` and doesn't output the address. Also warrants the question why shouldn't it include the address in the error from the store...?

Reply:
Yes, that may be useful. We will have to go through storage error messages and logging to clean and improve them.

Reply:
so @janos should we instead just add  the address to the error from the store?

Reply:
I think that it is ok to make this small improvement in this PR and have another one which would address the whole storage.